chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -540,3 +540,21 @@ def disable_if_exists(name):
                     break
 
         return error
+
+    @classmethod
+    def terminate_driver(cls, managed_mode):
+        # prune locks owned by THIS process
+        lockfiles = os.listdir(ZfsDevice.ZPOOL_LOCK_DIR)
+
+        for lockfile in lockfiles:
+            with open(ZfsDevice.ZPOOL_LOCK_DIR + lockfile, 'r') as f:
+                contents = f.readlines()
+
+            assert len(contents) == 1 and contents[0].isdigit(), \
+                'unexpected contents of lockfile %s: %s' % (lockfile, contents)
+
+            # validate pid holding lock is THIS process' pid
+            if os.getpid() == int(contents[0]):
+                os.remove(ZfsDevice.ZPOOL_LOCK_DIR + lockfile)
```

Comment:
So, all of this locking is to protect from having multiple zfs commands running concurrently, yes?

So if at this point in the shutdown there is a lock file present, does it not mean that there is a zfs command currently running as a child of the lock-holder?